kaolin/csrc/cuda/ops/conversions/unbatched_mcube/unbatched_mcube_cuda_kernel.cu

```diff
@@ -491,28 +495,52 @@ generateTriangles2(float *pos, int *faces, int *compactedVoxelArray,
     num_prev_verts = numPartialVertsScanned[target_voxel_idx3];
     num_prev_triangles = numTrianglesScanned[voxel];
 
-    faces[num_prev_triangles * 3 + j + 2] = num_prev_verts + offset3;
-  }
+    faces[num_prev_triangles * 3 + j] = num_prev_verts + offset3;
+    }
 }
 
 void launch_generateTriangles2(at::Tensor pos, at::Tensor faces, at::Tensor compactedVoxelArray,
-                              at::Tensor numTrianglesScanned, at::Tensor numPartialVertsScanned, at::Tensor numPartialVerts,
-                              at::Tensor voxelVertsOrder,
-                              int3 gridSize, at::Tensor voxelgrid,
-                              float3 voxelSize, float isoValue, int activeVoxels, int maxVerts)
+                                at::Tensor numTrianglesScanned, at::Tensor numPartialVertsScanned, at::Tensor numPartialVerts,
+                                at::Tensor voxelVertsOrder,
+                                int3 gridSize, at::Tensor voxelgrid,
+                                float3 voxelSize, float isoValue, int activeVoxels, int maxVerts)
 {
-  dim3 grid2((int) ceil(activeVoxels/ (float) NTHREADS), 1, 1);
+    dim3 grid2((int) ceil(activeVoxels/ (float) NTHREADS), 1, 1);
 
-  while (grid2.x > 65535) {
+    while (grid2.x > 65535) {
     grid2.x/=2;
     grid2.y*=2;
-  }
-
-  generateTriangles2<<<grid2, NTHREADS>>>(pos.data_ptr<float>(), faces.data_ptr<int>(),
-                                          compactedVoxelArray.data_ptr<int>(), numTrianglesScanned.data_ptr<int>(),
-                                          numPartialVertsScanned.data_ptr<int>(), numPartialVerts.data_ptr<int>(),
-                                          voxelVertsOrder.data_ptr<int>(),
-                                          voxelgrid.data_ptr<float>(), gridSize,
-                                          voxelSize, isoValue, activeVoxels,
-                                          maxVerts);
+    }
+
+    generateTriangles2<<<grid2, NTHREADS>>>(pos.data_ptr<float>(), faces.data_ptr<int>(),
+                                            compactedVoxelArray.data_ptr<int>(), numTrianglesScanned.data_ptr<int>(),
+                                            numPartialVertsScanned.data_ptr<int>(), numPartialVerts.data_ptr<int>(),
+                                            voxelVertsOrder.data_ptr<int>(),
+                                            voxelgrid.data_ptr<float>(), gridSize,
+                                            voxelSize, isoValue, activeVoxels,
+                                            maxVerts);
 }
+
+void CubScanWrapper(at::Tensor output, at::Tensor input, int numElements)
+{
+    int *d_in = input.data_ptr<int>();
+    int *d_out = output.data_ptr<int>();
+
+    cudaError_t err = cudaGetLastError();
+
+    if ( err != cudaSuccess )
+    {
+    printf("CUDA Error: %s\n", cudaGetErrorString(err));
+    exit(-1);
+    }
```

Comment:
What is the reason to put an error catching **before** executing the function ?

Reply:
For debbuging, fixed now.